mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1369,9 +1352,82 @@ def _compute_builtin_metrics(self):
                 )
             )
 
+    def _check_args(self, metrics, eval_df):
+        failed_metrics = []
+        # collect all failures for getting metric arguments
+        for metric in metrics:
+            result = self._get_args_for_metrics(metric, eval_df)
+            if isinstance(result, tuple):
+                failed_metrics.append(result)
+
+        if len(failed_metrics) > 0:
+            output_column_name = self.evaluator_config.get(
+                _Y_PREDICTED_OUTPUT_COLUMN_NAME, "output"
+            )
+            output_columns = (
+                [] if self.other_output_columns is None else list(self.other_output_columns.columns)
+            )
+            input_columns = list(self.X.copy_to_avoid_mutation().columns)
+
+            error_messages = []
+            for metric_name, param_names in failed_metrics:
+                error_messages.append(f"Metric '{metric_name}' requires the columns {param_names}")
+            error_message = "\n".join(error_messages)
+            raise MlflowException(
+                "Error: Metric calculation failed for the following metrics:\n"
+                f"{error_message}\n\n"
+                "Below are the existing column names for the input/output data:\n"
+                f"Input Columns: {input_columns}\n"
+                f"Output Columns: {output_columns}\n"
+                "Note that this does not include the output column: "
+                f"'{output_column_name}'\n\n"
+                f"To resolve this issue, you may want to map the missing column to an "
+                "existing column using the following configuration:\n"
+                f"evaluator_config={{'col_mapping': {{'<missing column name>': "
+                "'<existing column name>'}}\n"
+            )
+
+    def _evaluate_metrics(self, eval_df):
```

Comment:
im a bit confused about this function. if its purpose is checking the first row to make sure that the metric calculation works properly, can we name it as such? alternatively, can we generalize this function such that we can replace the full df calculation on L1426-1428 with this fn?

Reply:
yes, i can rename this function - the full df calculation has slightly different behavior than calculating the first row as it is now (it fails immediately when there is an error/exception raised, logs "Calculating x metric ..." etc.)